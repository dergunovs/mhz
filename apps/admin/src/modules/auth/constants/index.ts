export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
