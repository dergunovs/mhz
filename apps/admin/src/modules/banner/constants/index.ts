export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
