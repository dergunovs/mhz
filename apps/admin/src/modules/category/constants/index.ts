export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
