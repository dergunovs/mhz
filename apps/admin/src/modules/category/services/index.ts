import { ComputedRef } from 'vue';

import { api, useQuery, useMutation } from 'mhz-helpers';
import { API_CATEGORY, ICategory, IBaseReply } from 'mhz-contracts';

export function getCategories(options?: object) {
  async function fn() {
    const { data } = await api.get<{ data: ICategory[] }>(API_CATEGORY);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY], queryFn: fn, ...options });
}

export function getCategory(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: ICategory }>(`${API_CATEGORY}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY, id], queryFn: fn });
}

export function postCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.post<IBaseReply>(API_CATEGORY, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function updateCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.patch<IBaseReply>(`${API_CATEGORY}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function deleteCategory(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_CATEGORY}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}
