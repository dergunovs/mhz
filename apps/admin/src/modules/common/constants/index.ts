export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
