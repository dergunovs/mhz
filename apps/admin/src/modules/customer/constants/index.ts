export const URL_CUSTOMER = '/customers';

export const API_CUSTOMER = 'customer';
