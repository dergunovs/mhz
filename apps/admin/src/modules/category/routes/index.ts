import { URL_CATEGORY, URL_CATEGORY_CREATE, URL_CATEGORY_EDIT } from '@/category/constants';

export const categoryRoutes = [
  {
    path: URL_CATEGORY,
    name: 'CategoryList',
    component: () => import('@/category/pages/CategoryListPage.vue'),
  },
  {
    path: URL_CATEGORY_CREATE,
    name: 'CategoryCreate',
    component: () => import('@/category/pages/CategoryCreatePage.vue'),
  },
  {
    path: `${URL_CATEGORY_EDIT}/:id`,
    name: 'CategoryEdit',
    component: () => import('@/category/pages/CategoryEditPage.vue'),
  },
];
