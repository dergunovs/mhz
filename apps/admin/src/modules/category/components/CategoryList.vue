<template>
  <UiTable :headers="tableHeaders" :isLoading="!props.categories?.length">
    <template v-if="props.categories?.length">
      <tr v-for="category in props.categories" :key="category._id">
        <td data-grow>
          <RouterLink :to="`${URL_CATEGORY_EDIT}/${category._id}`">
            {{ category.title }}
          </RouterLink>
        </td>
        <td data-no-wrap>
          {{ formatDate(category.dateCreated) }}
        </td>
        <td data-no-wrap>
          {{ formatDate(category.dateUpdated) }}
        </td>
        <td>
          <UiButton @click="mutate(category._id)" layout="plain">Delete</UiButton>
        </td>
      </tr>
    </template>
  </UiTable>
</template>

<script setup lang="ts">
import { useQueryClient } from '@tanstack/vue-query';

import { ICategory } from 'mhz-types';
import { UiTable, UiButton, toast } from 'mhz-ui';
import { formatDate } from 'mhz-helpers';

import { deleteCategory } from '@/category/services';
import { API_CATEGORY, URL_CATEGORY_EDIT } from '@/category/constants';

interface IProps {
  categories?: ICategory[];
}

const props = defineProps<IProps>();

const tableHeaders = ['Category', 'Created', 'Updated', ''];

const queryClient = useQueryClient();

const { mutate } = deleteCategory({
  onSuccess: async () => {
    await queryClient.refetchQueries({ queryKey: [API_CATEGORY, 1], exact: true });
    toast.success('Category deleted');
  },
});
</script>
