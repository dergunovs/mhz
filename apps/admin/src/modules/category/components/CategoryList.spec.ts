import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import CategoryList from './CategoryList.vue';

import { wrapperFactory } from '@/common/test';
import { URL_CATEGORY_EDIT } from '@/category/constants';
import { CATEGORIES } from '@/category/fixtures';

let wrapper: VueWrapper;

const categoryListRow = '[data-test="category-list-row"]';
const categoryListLink = '[data-test="category-list-link"]';

beforeEach(() => {
  wrapper = wrapperFactory(CategoryList, {
    props: {
      categories: CATEGORIES.data,
    },
  });
});

enableAutoUnmount(afterEach);

describe('CategoryList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(CategoryList)).toBeTruthy();
  });

  it('shows rows if categories props', async () => {
    expect(wrapper.findAll(categoryListRow).length).toEqual(CATEGORIES.data.length);

    await wrapper.setProps({ categories: [] });

    expect(wrapper.find(categoryListRow).exists()).toBe(false);
  });

  it('generated link to category', async () => {
    const LINK = `${URL_CATEGORY_EDIT}/${CATEGORIES.data[0]._id}`;

    expect(wrapper.find(categoryListLink).attributes('to')).toEqual(LINK);
  });

  it('shows category content in cells', async () => {
    expect(wrapper.find(categoryListLink).text()).toEqual(CATEGORIES.data[0].title);
  });
});
