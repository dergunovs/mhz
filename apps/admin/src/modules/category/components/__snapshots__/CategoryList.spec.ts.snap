// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryList > matches snapshot 1`] = `
"<table headers="[object Object],[object Object],[object Object],[object Object]" isloading="false" lang="en">
  <tbody>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d4dc15f93f056da17ceff8" data-test="category-list-link">CPU</a></td>
      <td>327</td>
      <td data-no-wrap="">10 авг. 2023 г.</td>
      <td data-no-wrap="">21 дек. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d62fa7a635811931410ebd" data-test="category-list-link">Case</a></td>
      <td>28</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d62da6a635811931410ea4" data-test="category-list-link">Cooler</a></td>
      <td>15</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d62f06a635811931410eb3" data-test="category-list-link">GPU</a></td>
      <td>10</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d63066a635811931410ec7" data-test="category-list-link">Keyboard</a></td>
      <td>22</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d63016a635811931410ec2" data-test="category-list-link">Monitor</a></td>
      <td>5</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d626e88cde96b53dd37ff2" data-test="category-list-link">Motherboard</a></td>
      <td>19</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d630cea635811931410ecc" data-test="category-list-link">Mouse</a></td>
      <td>9</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d63106a635811931410ed1" data-test="category-list-link">Mousepad</a></td>
      <td>5</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d62f6ca635811931410eb8" data-test="category-list-link">PSU</a></td>
      <td>6</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d62e42a635811931410ea9" data-test="category-list-link">RAM</a></td>
      <td>4</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
    <tr data-test="category-list-row">
      <td data-grow=""><a to="/categories/edit/64d62ea8a635811931410eae" data-test="category-list-link">SSD</a></td>
      <td>8</td>
      <td data-no-wrap="">11 авг. 2023 г.</td>
      <td data-no-wrap="">20 сент. 2023 г.</td>
    </tr>
  </tbody>
</table>"
`;
